src/k8s/pkg/k8sd/features/calico/cleanup.go

```diff
@@ -0,0 +1,81 @@
+package calico
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"golang.org/x/sys/unix"
+)
+
+func CleanupNetwork(ctx context.Context, snap snap.Snap) error {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return fmt.Errorf("failed to list network interfaces: %w", err)
+	}
+
+	// Find the interfaces created by Calico
+	for _, iface := range interfaces {
+		// Check if the interface name matches the regex pattern
+		match, err := regexp.MatchString("^vxlan[-v6]*.calico|cali[a-f0-9]*$", iface.Name)
+		if err != nil {
+			return fmt.Errorf("failed to match regex pattern: %w", err)
+		}
+		if match {
+			// Perform cleanup for Calico interface
+			if err := exec.CommandContext(ctx, "ip", "link", "delete", iface.Name).Run(); err != nil {
+				return fmt.Errorf("failed to delete interface %s: %w", iface.Name, err)
+			}
+		}
+	}
+
+	// Delete network namespaces that start with "cali-"
+	netnsDir := "/run/netns"
+	entries, err := os.ReadDir(netnsDir)
+	if err != nil {
+		return fmt.Errorf("failed to list files under %s: %w", netnsDir, err)
+	}
+
+	for _, entry := range entries {
+		if strings.HasPrefix(entry.Name(), "cali-") {
+			nsPath := path.Join(netnsDir, entry.Name())
+
+			if err := unix.Unmount(nsPath, unix.MNT_DETACH); err != nil {
+				return fmt.Errorf("failed to unmount network namespace %s: %w", entry.Name(), err)
+			}
+
+			if err := os.Remove(nsPath); err != nil {
+				return fmt.Errorf("failed to remove network namespace %s: %w", entry.Name(), err)
+			}
+		}
+	}
+
+	for _, cmd := range []string{"iptables", "ip6tables", "iptables-legacy", "ip6tables-legacy"} {
+		out, err := exec.Command(fmt.Sprintf("%s-save", cmd)).Output()
+		if err != nil {
+			return fmt.Errorf("failed to read iptables rules: %w", err)
+		}
+
+		grep := exec.Command("grep", "-iv", "cali")
+		grep.Stdin = bytes.NewReader(out)
+		grepOut, err := grep.Output()
+		if err != nil {
+			return fmt.Errorf("failed to filter calico iptables rules: %w", err)
+		}
```

Comment:
this should not be a command. just iterate over the output lines and drop the ones that do not match